src/FiberTraceSet.cc

```diff
@@ -181,7 +181,8 @@ SpectrumSet FiberTraceSet<ImageT, MaskT, VarianceT>::extractSpectra(
                 if (modelValue > minFracMask) {
                     maskResult[ii] |= maskValue;
                 }
-                if ((maskValue & badBitMask) || !std::isfinite(imageValue) || !std::isfinite(varianceValue)) {
+                if ((maskValue & badBitMask) || !std::isfinite(imageValue) || !std::isfinite(varianceValue) ||
+                    varianceValue == 0) {
```

Comment:
Is `varianceValue` never negative?

Reply:
It should never be zero either, so probably worth making this `<=`.